src/features/channels/components/channel-list/channel-list-item/ChannelListItem.tsx

```diff
@@ -31,16 +33,42 @@ const ChannelListItem = ({
   selected = false,
 }: Props) => {
   const classes = useStyles();
-  const { mutedChannels } = useAppSelector((state) => state.channel);
+  const {
+    channel: { mutedChannels },
+    auth: {
+      userData: { privateKeyArmored, passphrase },
+    },
+  } = useAppSelector((state) => state);
   const isMuted = mutedChannels.includes(channel.channelAddress);
   const { onSendMessage } = useChat(channel.channelAddress);
   const dispatch = useAppDispatch();
 
   // When a new message is created we add that message to the 'messages' property
   // of the channel (this is a socket event so we only connect to it on mount)
   useOnMount(() => {
-    onSendMessage((message) => {
-      dispatch(addNewMessage({ channelAddress: channel.channelAddress, message }));
+    onSendMessage(async (message) => {
+      const encryptionService = new EncryiptionService();
+
+      dispatch(
+        addNewMessage({
+          channelAddress: channel.channelAddress,
+          message: {
+            ...message,
+            decryptedMessage: await encryptionService.decryptMessage(
+              message.message,
+              passphrase,
+              privateKeyArmored
+            ),
+            decryptedReplyMessage:
+              message.replyMessage &&
+              (await encryptionService.decryptMessage(
```

Comment:
This is very hard to understand, I'd suggest to move most of this logic to actions, or put it into a variable and reference it from here